src/ocm/hooks/useInfraEnvId.ts

```diff
@@ -3,16 +3,18 @@ import { Cluster, InfraEnv } from '../../common';
 import { APIErrorMixin } from '../api/types';
 import { InfraEnvsService } from '../services';
 
+type ErrorType = APIErrorMixin & Error;
+
 export default function useInfraEnvId(clusterId: Cluster['id']) {
   const [infraEnvId, setInfraEnv] = React.useState<InfraEnv['id']>();
-  const [error, setError] = React.useState<APIErrorMixin & Error>();
+  const [error, setError] = React.useState<ErrorType>();
 
   const findInfraEnvId = React.useCallback(async () => {
     try {
       const infraEnvId = await InfraEnvsService.getInfraEnvId(clusterId);
       setInfraEnv(infraEnvId);
     } catch (e) {
-      setError(e);
+      setError(e as ErrorType);
```

Comment:
And here as well. Looking at the usage the error is basically only used as a boolean (if (error) ...) so I think it would be better to just store the error message?